### PR TITLE
improve: replace jenvtest with kube-api-test from fabric8 client

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

File: operator-framework/pom.xml [hunk 1]
```diff
@@ -85,9 +85,8 @@
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>io.javaoperatorsdk</groupId>
-      <artifactId>jenvtest</artifactId>
-      <scope>test</scope>
```
Comment (metacosm) [inline operator-framework/pom.xml:90]: Shouldn't it still use the test scope?

Comment (csviri) [reply operator-framework/pom.xml:90]: I guess that is inherited from dependency management

Comment (metacosm) [reply operator-framework/pom.xml:90]: https://stackoverflow.com/questions/15221299/dependencymanagement-and-scope seems to indicate that putting scopes in `dependencyManagement` is a bad practice…

Comment (csviri) [reply operator-framework/pom.xml:90]: hmm, ok np, will move it

**Review by metacosm — APPROVED** [review]

**Review by metacosm — APPROVED** [review]

**Pull request merged**